scripts/src/java/org/oppia/android/scripts/RegexPatternValidationCheck.kt

```diff
@@ -0,0 +1,232 @@
+     */
+    fun checkProhibitedContent(
+      repoPath: String,
+      searchFiles: Sequence<File>,
+      fileNameRegexString: String,
+      prohibitedContentRegexString: String,
+      errorToShow: String
+    ): Boolean {
+      var contentCheckFailedFlag = false
+      val fileNameRegex = fileNameRegexString.toRegex()
+      val prohibitedContentRegex = prohibitedContentRegexString.toRegex()
+      searchFiles.forEach {
+        if (fileNameRegex.matches(it.name)) {
+          var lineNumber = 0
+          File(it.toString()).forEachLine { lineString ->
+            lineNumber++
+            if (prohibitedContentRegex.matches(lineString)) {
+              logProhibitedContentFailure(
+                lineNumber = lineNumber,
+                errorToShow = errorToShow,
+                filePath = it.toString().removePrefix(repoPath)
+              )
+              contentCheckFailedFlag = true
+            }
+          }
+        }
+      }
+      return contentCheckFailedFlag
+    }
+
+    /**
+     * Collects the paths of all the files which are needed to be checked
+     *
+     * @param repoPath the path of the repo.
+     * @param allowedDirectories a list of all the directories which needs to be checked.
+     * @param exemptionList a list of files which needs to be exempted for this check
+     * @return [Sequence<File>] all files which needs to be checked.
+     */
+    fun collectSearchFiles(
+      repoPath: String,
+      allowedDirectories: MutableList<String>,
+      exemptionList: Array<String> = arrayOf<String>()
+    ): Sequence<File> {
+      val validPaths = File(repoPath).walk().filter { it ->
+        checkIfAllowedDirectory(
+          it.toString().removePrefix(repoPath),
+          allowedDirectories)
+          && it.isFile
+          && it.name !in exemptionList
+      }
+      return validPaths
+    }
+
+    /**
+     * Checks if a layer is allowed to be analyzed for the check or not.
+     * It only allows the layers listed in allowedDirectories
+     * (which is specified from the command line arguments) to be analyzed.
+     *
+     * @param pathString the path of the repo.
+     * @param allowedDirectories a list of all the files which needs to be checked.
+     * @return [Boolean] check failed or passed
+     */
+    fun checkIfAllowedDirectory(
+      pathString: String,
+      allowedDirectories: MutableList<String>
+    ): Boolean {
+      allowedDirectories.forEach {
+        if (pathString.startsWith(it))
+          return true
+      }
+      return false
+    }
+
+    /** Logs the failures for filename pattern violation
+     *
+     * @param errorToShow the failure message to be logged
+     * @param filePath the path of the file relative to the repository which failed the check
+     */
+    fun logProhibitedFilenameFailure(
+      errorToShow: String,
+      filePath: String
+    ) {
+      println("Prohibited filename pattern: [ROOT]/$filePath\n" +
+        "Failure message: $errorToShow\n")
```

Comment:
```suggestion
      println(
        "Prohibited filename pattern: [ROOT]/$filePath\nFailure message: $errorToShow\n"
      )
```

Here & elsewhere: prefer breaking at the higher syntactal element (e.g. the ``(`` here) before the lower element (the string).

Reply:
okay, noted.